src/js/dp/search.js

```diff
@@ -8,138 +8,173 @@ const fetchHtml = async (url) => {
     }
   }
-}
+};
 
 const switchCheckbox = (paramsArray) => {
   // get current param
-  let strParams = window.location.search; 
+  let strParams = window.location.search;
 
   // build new param
   paramsArray.map((param) => {
-    if(!('isChecked' in param) || !('filterName' in param)) return;
+    if (!("isChecked" in param) || !("filterName" in param)) return;
     if (param.isChecked) {
       strParams += `&filter=${param.filterName}`;
     } else {
-      strParams = strParams.replace(new RegExp(`(\\&|\\?)filter\=${param.filterName}`), "")
+      strParams = strParams.replace(
+        new RegExp(`(\\&|\\?)filter\=${param.filterName}`),
+        ""
+      );
     }
   });
 
   // make the change to the markup
-  switchSearchMarkup(strParams, true)
-}
+  switchSearchMarkup(strParams, true);
+};
 
-// create listeners for checkboxes controlling eachother
-[...searchContainer.querySelectorAll('.ons-checkboxes__items [aria-controls]')].map(topFilter => {
-  const childrenSelector = topFilter.getAttribute('aria-controls');
-  const theChildren = [...searchContainer.querySelectorAll(`#${childrenSelector} [type=checkbox]`)];
-  if(!childrenSelector) return;
-  topFilter.addEventListener("change", async (e) => {
-      const paramsArray = theChildren.map(item => ({isChecked: e.target.checked, filterName: item.value}));
-      theChildren.map(item => item.checked = e.target.checked);
+if (searchContainer) {
+  // create listeners for checkboxes controlling eachother
+  [
+    ...searchContainer.querySelectorAll(
+      ".ons-checkboxes__items [aria-controls]"
+    ),
+  ].map((topFilter) => {
+    const childrenSelector = topFilter.getAttribute("aria-controls");
+    const theChildren = [
+      ...searchContainer.querySelectorAll(
+        `#${childrenSelector} [type=checkbox]`
+      ),
+    ];
+    if (!childrenSelector) return;
+    topFilter.addEventListener("change", async (e) => {
+      const paramsArray = theChildren.map((item) => ({
+        isChecked: e.target.checked,
+        filterName: item.value,
+      }));
+      theChildren.map((item) => (item.checked = e.target.checked));
       switchCheckbox(paramsArray);
-  });
-  theChildren.map((item) => {
+    });
+    theChildren.map((item) => {
       item.addEventListener("change", async (e) => {
-        switchCheckbox([{isChecked: e.target.checked, filterName: e.target.value}]);
-        topFilter.checked = theChildren.some(x => x.checked);
-      })
-  });    
-});
-
-// create listeners for the sort dropdown
-const sortSelector = searchContainer.querySelector(".ons-input--sort-select");
-if(!!sortSelector) {
-  sortSelector.addEventListener("change", async (e) => {
-    let strParams = window.location.search;
-    // remove old param
-    strParams = strParams.replace(new RegExp(`[?&]sort\=[^&]+`), "")
-    // replace
-    strParams += `&sort=${e.target.value}`;
-    switchSearchMarkup(strParams, true)
+        switchCheckbox([
```

Comment:
`async` unnecessary because it doesn't `await` anything or need to implicitly return a promise.